src/containerapp/azext_containerapp/custom.py

```diff
@@ -4297,3 +4306,195 @@ def delete_workload_profile(cmd, resource_group_name, env_name, workload_profile
         return r
     except Exception as e:
         handle_raw_exception(e)
+
+
+def patch_list(cmd, resource_group_name=None, managed_env=None, show_all=False):
+    if is_docker_running() is False:
+        logger.warning("Please install or start Docker and try again.")
+        return
+    pack_exec_path = get_pack_exec_path()
+    print("\rListing container apps...", end="", flush=True)
```

Comment:
@snehapar9 @harryli0108 -- we're using `print()` in a lot of places where we should be using `logger.warning`, `logger.info` or `logger.debug` depending on the type of stream we want to write to (I know `logger.warning` and `logger.error` show by default, I _think_ `logger.info` shows up with `--verbose` and `logger.debug` shows up with `--debug`)

Reply:
The print statements with `end=""` is for prompting users with updates. These print statements can rewrite the previous print lines by using the `\r` in their statement without writing too many lines. I'm not sure if logger can allow these behaviors

Reply:
Updated: changed some print statement to logger methods.